modules/rpc/yarpc.go

```diff
@@ -212,39 +219,61 @@ func newYARPCModule(
 	module := &YARPCModule{
 		ModuleBase: *modules.NewModuleBase(name, mi.Host, []string{}),
 		register:   reg,
-		config:     yarpcConfig{AdvertiseName: mi.Host.Name()},
 	}
 
 	module.log = ulog.Logger().With("moduleName", name)
 	for _, opt := range options {
 		if err := opt(&mi); err != nil {
-			module.log.Error("Unable to apply option", "error", err, "option", opt)
 			return module, errs.Wrap(err, "unable to apply option to YARPC module")
 		}
 	}
 
 	config := module.Host().Config().Scope("modules").Get(module.Name())
 	if err := config.PopulateStruct(&module.config); err != nil {
-		return module, err
+		return nil, errs.Wrap(err, "can't read inbounds")
 	}
 
+	// iterate over inbounds
+	transportsIn, err := prepareInbounds(module.config.Inbounds, mi.Host.Name())
+	if err != nil {
+		return nil, errs.Wrap(err, "can't process inbounds")
+	}
+
+	module.config.transports.inbounds = transportsIn
 	module.config.inboundMiddleware = inboundMiddlewareFromCreateInfo(mi)
 	module.config.onewayInboundMiddleware = onewayInboundMiddlewareFromCreateInfo(mi)
 
-	tchTransport, err := tch.NewChannelTransport(
-		tch.ServiceName(module.config.AdvertiseName),
-		tch.ListenAddr(module.config.Bind),
-	)
+	_controller.addConfig(module.config)
```

Comment:
Won't this cause duplicates since addDefaultMiddleware has already added the yarpcConfig to c.configs?